server/shared/models/query_builder.js

```diff
@@ -29,6 +29,28 @@ const { curry, pipe } = require('ramda');
  * @param   {QueryObject} queryObj
  * @returns {QueryObject}
  */
+
+const addInnerJoinClause = curry(({ innerJoin }, queryObj) => {
```

Comment:
I think specifying the join as an option makes it more opaque (see your select query in `visitors_all`). When we join we're creating a new temporary table, so actually, it's the first argument to `select` that should be different.

I think I like this:
```js
selectQuery(
  join('users', { visits: 'user_id', activities: 'user_id' }),
  [ ... ],
  { where: { foo: 'bar' } },
)
```
Where the first argument is the base table, and the object holds table - join-on pairs.

What do you think?